core/src/main/java/uk/gov/gchq/koryphe/impl/predicate/If.java

```diff
@@ -0,0 +1,163 @@
+    private Predicate<I> otherwise;
+
+    public If() {
+    }
+
+    public If(final Boolean condition, final Predicate<I> then, final Predicate<I> otherwise) {
+        this.condition = condition;
+        this.then = then;
+        this.otherwise = otherwise;
+    }
+
+    public If(final Predicate<I> predicate, final Predicate<I> then, final Predicate<I> otherwise) {
+        this.predicate = predicate;
+        this.then = then;
+        this.otherwise = otherwise;
+    }
+
+    public Boolean getCondition() {
+        return condition;
+    }
+
+    public void setCondition(final Boolean condition) {
+        this.condition = condition;
+    }
+
+    public Predicate<I> getThen() {
+        return then;
+    }
+
+    public void setThen(final Predicate<I> then) {
+        this.then = then;
+    }
+
+    public Predicate<I> getOtherwise() {
+        return otherwise;
+    }
+
+    public void setOtherwise(final Predicate<I> otherwise) {
+        this.otherwise = otherwise;
+    }
+
+    public Predicate<I> getPredicate() {
+        return predicate;
+    }
+
+    public void setPredicate(final Predicate<I> predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public boolean test(final I input) {
+        final boolean computedCondition;
+        if (null == condition) {
+            computedCondition = null != predicate && predicate.test(input);
+        } else {
+            computedCondition = condition;
+        }
+
+        if (computedCondition) {
+            return null != then && then.test(input);
+        }
+        return null != otherwise && otherwise.test(input);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (null == obj || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        final If ifPredicate = (If) obj;
+
+        return new EqualsBuilder()
+                .append(condition, ifPredicate.condition)
+                .append(predicate, ifPredicate.predicate)
+                .append(then, ifPredicate.then)
+                .append(otherwise, ifPredicate.otherwise)
+                .build();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder(11, 73)
+                .append(condition)
+                .append(predicate)
+                .append(then)
+                .append(otherwise)
+                .build();
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this)
+                .append("condition", condition)
+                .append("predicate", predicate)
+                .append("then", then)
+                .append("otherwise", otherwise)
+                .toString();
+    }
+
+    @Override
+    public ValidationResult isInputValid(final Class<?>... args) {
+        if (null == predicate || null == then || null == otherwise) {
```

Comment:
Perhaps you could do something like:
```java
final ValidationResult result = new ValidationResult();
if(null != predicate) {
    result.add(Signature.getInputSignature(predicate).assignable(args));
}
if(null != then) {
    result.add(Signature.getInputSignature(then).assignable(args));
}
if(null != otherwise) {
    result.add(Signature.getInputSignature(otherwise).assignable(args));
}
return result
```